### PR TITLE
Release helm chart

### LINKED ISSUE #882: Official helm chart 🎉
I'm currently writing a contribution and looking to help maintain a localstack incubator chart within helm/charts for the localstack community. They're is already an [unofficial project](https://github.com/everpeace/helm-charts/tree/master/localstack) which can be improved upon with permission of the author and a few interested contributors - please get in touch if you're still interested! 

There are some requirements to making the chart stable which would need to be discussed with maintainers of localstack organisation. For instance, the [technical requirements](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements) specify all docker image source code needs to be in Github repositories. I've hit a problem that the dependent image (`localstack/java-maven-node-python`) is not available. After inspecting the image locally it seems to be relatively simple and I was wondering whether you could push this and setup integrations for security screening and github on your docker hub account please? In return I will gladly look at using a [multistage build](https://docs.docker.com/v17.09/engine/userguide/eng-image/multistage-build/) to reduce the image size further.

Also, it is preferred to use fixed versions rather than a latest tag within helm conventions so the way latest is tagged without another reference is likely to cause a conflict when trying to get a recently patched build.

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-313) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by mforutan**
Using GitHub Actions to release the official helm chart

This PR is a starting point to introduce an official helm chart for locakstack,

I copied the helm chart from https://github.com/sidhuko/charts/pull/1 without any modification (hope it is fine with @sidhuko)

There are couple of tasks that need to be done before this can be merged:
- A `gh-pages` branch need to be created
- `GitHub Pages` need to be enabled in the setting

TODO (can be done in this PR or next PRs):
- A GitHub workflow to test the helm chart for PRs

Fixes #882

### CODE REVIEW

File: chart/localstack/values.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    - name: cloudwatch
+      enabled: false
+      internalPort: 4582
+      externalPort: 4582
+      nodePort: 31582
+    - name: secretsmanager
+      enabled: false
+      internalPort: 4583
+      externalPort: 4583
+      nodePort: 31583
```
Comment (macnev2013) [inline chart/localstack/values.yaml:111]: We can remove this except `4566` and `4571`.

**Review by macnev2013 — CHANGES_REQUESTED** [review]

File: chart/localstack/values-production.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    - name: cloudwatch
+      enabled: false
+      internalPort: 4582
+      externalPort: 4582
+      nodePort: 31582
+    - name: secretsmanager
+      enabled: false
+      internalPort: 4583
+      externalPort: 4583
+      nodePort: 31583
```
[the remark this answers is not in the document]
Comment (sidhuko) [reply chart/localstack/values-production.yaml:111]: How about setting <SERVICE>_PORT_EXTERNAL or is this not required anymore?

**Comment by sidhuko** [comment]
Sorry, been a busy week. I've created a new branch (localstack-chart) to submit as a helm chart on their repository but it needs some work. I am unsure about contributors comments above as it seems we still have `<SERVICE>_PORT_EXTERNAL` and `<SERVICE>_BACKEND`. I am not sure if both are still used? I've supported `<SERVICE>_BACKEND` which can be specified by passing an key value pair under `services` of the new configuration in new branch. If neither are support or will be deprecated it will make it easier. Otherwise I can still expose multiple ports by reverting to what we have here.

We will also need to expose the new environment variables as configuration in the deployment and values file.

File: chart/localstack/values-production.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (macnev2013) [reply chart/localstack/values-production.yaml:111]: Yes, as per my knowledge it is deprecated. We are using single entrypoint for all the services (by default `4566`) and 4571.

Ref. https://github.com/localstack/localstack#announcements
Ref. [docker-compose.yaml](https://github.com/localstack/localstack/blob/master/docker-compose.yml#L8-L10)

**Comment by macnev2013** [comment]
> Sorry, been a busy week. I've created a new branch (localstack-chart) to submit as a helm chart on their repository but it needs some work. I am unsure about contributors comments above as it seems we still have `<SERVICE>_PORT_EXTERNAL` and `<SERVICE>_BACKEND`. I am not sure if both are still used? I've supported `<SERVICE>_BACKEND` which can be specified by passing an key value pair under `services` of the new configuration in new branch. If neither are support or will be deprecated it will make it easier. Otherwise I can still expose multiple ports by reverting to what we have here.

Hey @sidhuko, As far as I know we don't use it anymore. @whummer can you confirm it?

**Comment by macnev2013** [comment]
> We will also need to expose the new environment variables as configuration in the deployment and values file.

Yes @sidhuko, we need to add new variables as well. For the starting point maybe we can start importing variables present in the compose file.

Ref. [docker-compose.yaml](https://github.com/localstack/localstack/blob/master/docker-compose.yml#L12-L18)

**Comment by mforutan** [comment]
@macnev2013 @sidhuko, what do you think if we change the chart's version to `0.1.0-beta.0` and merge this PR to have the process of releasing the chart in place then we can fix the rest of issues and release the chart in next PRs, my aim here is to keep these two steps separated so everyone can contribute individually.

**Comment by whummer** [comment]
Thanks @mforutan for creating the initial chart version 👍 and thanks @macnev2013 @sidhuko for reviewing. Agreed that we can merge this initial version as-is, we'll make a couple of changes and enhancements (e.g., removing the legacy service ports) as we go.. Thanks

**Comment by mforutan** [comment]
hi @whummer, thank you for merging this PR but it failed to deploy the chart as the `gh-pages` branch wasn't created, can you please do these steps to get the chart deployed?

- Remove the new tag: https://github.com/localstack/localstack/releases/tag/localstack-0.1.0
- Create `gh-pages` branch
- Enable GitHub pages in the setting
- Re-run the GitHub Actions: https://github.com/localstack/localstack/actions/runs/405281754

**Comment by whummer** [comment]
Thanks @mforutan . Actually, I think we may need to move the chart to a separate repo in that case, or do you know if there is a way to prevent the `helm/chart-releaser-action@v1.1.0` Github action plugin from creating tags (since we're already using tags to create LocalStack code releases)?

**Comment by mforutan** [comment]
[helm/chart-releaser](https://github.com/helm/chart-releaser) uses GitHub release to store the chart and also to compare it with the existing versions, 

If you are able to create a new repo, I am more than happy to create a new PR against that or we may need to see how to ignore this new tags in code releases.

File: chart/localstack/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+1. Get the application URL by running these commands:
+{{- if .Values.ingress.enabled }}
+{{- range $host := .Values.ingress.hosts }}
+  {{- range $.Values.ingress.paths }}
```
Comment (JamesArthurHolland) [inline chart/localstack/templates/NOTES.txt:4]: Error: template: templates/NOTES.txt:4:12: executing "templates/NOTES.txt" at <.paths>: can't evaluate field paths in type interface {}

values.yml - ingress
ingress:
  enabled: true
  annotations: {}
    # kubernetes.io/ingress.class: nginx
    # kubernetes.io/tls-acme: "true"
  paths:
    - /
  hosts:
    - chart-example.local
  tls: []
  #  - secretName: chart-example-tls
  #    hosts:
  #      - chart-example.local

Comment (JamesArthurHolland) [reply chart/localstack/templates/NOTES.txt:4]: The error is there also when paths isn't altered from the PR

File: chart/localstack/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      nodePort: {{ .nodePort }}
+      {{- end }}
+      {{- end }}
+  {{- range $index, $val := .Values.localstack.services }}
+  {{- if $val.enabled }}
+    - name: {{ $val.name }}
+      protocol: TCP
+      port: {{ $val.externalPort }}
+      targetPort: {{ $val.internalPort }}
+      {{- if contains "NodePort" .Values.service.type }}
```
Comment (JamesArthurHolland) [inline chart/localstack/templates/service.yaml:28]: This creates a scope error:
`Error: template: localstack/templates/service.yaml:28:40: executing "localstack/templates/service.yaml" at <.Values.service.type>: nil pointer evaluating interface {}.service`


I think it should be 
`$.Values.service.type`

**Comment by sidhuko** [comment]
I cannot say as not a contributor here. It's early Monday morning for me. I
will hopefully complete by end of day today and get the PR across to
helm/charts.

They are quite responsive. Merge in couple days as long as no significant
changes are needed. I will leave notes here so you can try install locally
and test for me. The chart will be better in helm/charts then here for
continued support though.

**Comment by whummer** [comment]
Thanks all for your inputs - we'll first migrate the chart into a separate repo later today, then we can continue making the incremental changes. Will share the details here shortly. Thanks

**Comment by whummer** [comment]
@mforutan @sidhuko @macnev2013 @JamesArthurHolland Moved the chart into this new repo: https://github.com/localstack/helm-charts . We'll create the gh-pages branch and the Github actions build shortly.. Thanks

**Comment by mforutan** [comment]
@whummer Awesome!

**Pull request merged**